controllers/workloads/cfbuild_controller.go

```diff
@@ -90,25 +94,64 @@ func (r *CFBuildReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
+		// Scenario: CFBuild newly created and all status conditions are unknown, it
+		// Creates a KpackImage resource to trigger staging.
+		// Updates status on CFBuild -> sets staging to True.
 		err = r.createKpackImageAndUpdateStatus(ctx, &cfBuild, &cfApp, &cfPackage)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
+	} else if stagingStatus == metav1.ConditionTrue &&
+		readyStatus == metav1.ConditionUnknown &&
+		succeededStatus == metav1.ConditionUnknown {
+		// Scenario: CFBuild reconciles when Type staging is True and Type ready is False, it
+		// Retrieves and Checks Kpack Image Status Condition for Type "Succeeded"
+		// If NotFound error - Ignore and return
+		// If Found, check Succeeded status condition
+		// If Succeeded is True - Update Status Conditions and Droplet fields on CFBuild
+		// If Succeeded is False - Update Status Conditions on CFBuild
+		var kpackImage buildv1alpha1.Image
+		err = r.Client.Get(ctx, types.NamespacedName{Name: cfBuild.Name, Namespace: cfBuild.Namespace}, &kpackImage)
+		if err != nil {
+			r.Log.Error(err, "Error when fetching Kpack Image")
+			//Ignore Image NotFound errors to account for eventual consistency
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+		kpackSucceededStatusCondition := kpackImage.Status.GetCondition(kpackReadyConditionType)
+		if kpackSucceededStatusCondition.IsFalse() {
+			// Set CFBuild status Conditions on local copy - Staging and Succeeded to False
+			setStatusConditionOnLocalCopy(&cfBuild.Status.Conditions, workloadsv1alpha1.StagingConditionType, metav1.ConditionFalse, "kpack", "kpack")
+			setStatusConditionOnLocalCopy(&cfBuild.Status.Conditions, workloadsv1alpha1.SucceededConditionType, metav1.ConditionFalse, "kpack", "kpack")
+			if err := r.Client.Status().Update(ctx, &cfBuild); err != nil {
+				r.Log.Error(err, "Error when updating CFBuild status")
+				return ctrl.Result{}, err
+			}
+		} else if kpackSucceededStatusCondition.IsTrue() {
+			// Set CFBuild status Conditions on local copy- Staging to False and Succeeded to True
+			setStatusConditionOnLocalCopy(&cfBuild.Status.Conditions, workloadsv1alpha1.StagingConditionType, metav1.ConditionFalse, "kpack", "kpack")
+			setStatusConditionOnLocalCopy(&cfBuild.Status.Conditions, workloadsv1alpha1.SucceededConditionType, metav1.ConditionTrue, "kpack", "kpack")
+			// Generate Droplet object using Kpack Image and set it on CFBuild local copy
+			cfBuild.Status.BuildDropletStatus = r.generateBuildDropletStatus(&kpackImage)
+			// Call Status().Update() tp push updates to the server
+			if err := r.Client.Status().Update(ctx, &cfBuild); err != nil {
+				r.Log.Error(err, "Error when updating CFBuild status")
+				return ctrl.Result{}, err
+			}
+		}
 	}
-
 	return ctrl.Result{}, nil
 }
 
 func (r *CFBuildReconciler) createKpackImageAndUpdateStatus(ctx context.Context, cfBuild *workloadsv1alpha1.CFBuild, cfApp *workloadsv1alpha1.CFApp, cfPackage *workloadsv1alpha1.CFPackage) error {
```

Comment:
I see internally this calls `createKpackImageIfNotExists` -- what is the behavior if a kpackImage does exist? Should we be calling a `createOrUpdate` type method instead?

Reply:
kpackImage "exists" condition is encountered when we were successful in creating `kpack image` but encountered an error while updating the status on the `cfbuild`. Hence, when we reconcile again, we continue from where we left and just update the status. 
What would have changed on the CFBuild that would be valuable to run an update? 

`createOrUpdate` is not part of `client.Client` pkg. CFClient interface we have defined will have to change to accommodate for it.